cocotb_build_libs.py

```diff
@@ -363,24 +368,37 @@ def get_ext():
     else:
         modelsim_dir = os.path.dirname(os.path.dirname(vsim_path))
         modelsim_include_dir = os.path.join(modelsim_dir, "include")
+        modelsim_lib_path = os.path.dirname(vsim_path)
+        if os.name == "nt":
+            modelsim_libmtipli_path = os.path.join(modelsim_lib_path, "mtipli.dll")
+        else:
+            modelsim_libmtipli_path = os.path.join(modelsim_lib_path, "libmtipli.so")
         mti_path = os.path.join(modelsim_include_dir, "mti.h")
         if os.path.isfile(mti_path):
-            lib_name = "libcocotbfli_modelsim"
-            fli_ext = Extension(
-                os.path.join("cocotb", "libs", lib_name),
-                include_dirs=[include_dir, modelsim_include_dir],
-                libraries=["gpi", "gpilog", "stdc++"] + modelsim_extra_lib,
-                library_dirs=modelsim_extra_lib_path,
-                sources=[
-                    os.path.join(share_lib_dir, "fli", "FliImpl.cpp"),
-                    os.path.join(share_lib_dir, "fli", "FliCbHdl.cpp"),
-                    os.path.join(share_lib_dir, "fli", "FliObjHdl.cpp"),
-                ],
-                extra_link_args=_extra_link_args(lib_name=lib_name, rpaths=["$ORIGIN"]),
-                extra_compile_args=_extra_cxx_compile_args,
-            )
+            if _can_load_library(modelsim_libmtipli_path):
```

Comment:
Not sure if `_can_load_library` is good idea. If I am not mistaken (I can easily be) we do not need to load libraries here (on POSIX this would mean `libmtipli.so` is available to load which may not be the case at this point on Windows `modelsim_lib_path` is on `PATH`). If `mti.h` is available we can probably assume "dll/so" also. Alternative just check `os.path.isfile`?

Reply:
This is not a about checking whether the file is present but rather if it is compatible. If 32-bit modelsim is installed and a 64-bit python environment is used `pip install` results in a linker error because the 32-bit `libmtipli` cannot be linked with the 64-bit `libcocotbfli_modelsim`. Rather than stoping with a linker error I prefer to throw a warning and skip `libcocotbfli_modelsim`. `libmtipli.so` is loaded from the folder which also contains `vdbg`. An absolute path is used and therefore it should not be necessary to have different `LD_LIBRARY_PATH`. It works for me on Windows and Linux.

Reply:
OK now I get it. In this case probbaly a line of comment would be good.